lib/application/widgets/theme/tags_field.dart

```diff
@@ -0,0 +1,273 @@
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter_hooks/flutter_hooks.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+import 'package:layoutr/common_layout.dart';
+import 'package:memo/application/constants/animations.dart' as anims;
+import 'package:memo/application/constants/dimensions.dart' as dimens;
+import 'package:memo/application/constants/images.dart' as images;
+import 'package:memo/application/constants/strings.dart' as strings;
+import 'package:memo/application/hooks/tags_controller_hook.dart';
+import 'package:memo/application/theme/theme_controller.dart';
+
+/// Provides the list of tags of the collection being updated.
+///
+/// These tags may be mutated by [TagsField], which the user may use to update the tags list.
+class TagsController extends ValueNotifier<List<String>> {
+  TagsController({List<String>? tags}) : super(tags ?? []);
+
+  List<String> get tags => value;
+
+  set tags(List<String> tags) => value = List.from(tags);
+}
+
+/// A custom [TextField] that groups a collection of tags.
+class TagsField extends HookConsumerWidget {
+  const TagsField({this.controller, this.maxTags = 5, this.errorText});
+
+  /// Controls the tags being edited.
+  ///
+  /// If null, this widget will create its own [TagsController].
+  final TagsController? controller;
+
+  /// {@template TagsDropdownField.maxTags}
+  /// Maximum number of tags allowed to be chosen.
+  ///
+  /// Inputs that would increase this amount will be ignored, i.e, not transformed in tags.
```

Comment:
Not sure exactly the meaning of this

Reply:
Removed the second sentence because it didn't added any relevant information